pkgs/development/libraries/mesa/default.nix

```diff
@@ -9,7 +9,7 @@
 , vulkanDrivers ? ["auto"]
 , eglPlatforms ? [ "x11" ] ++ lib.optionals stdenv.isLinux [ "wayland" ]
 , OpenGL, Xplugin
-, withValgrind ? !stdenv.isDarwin && lib.meta.availableOn stdenv.hostPlatform valgrind-light, valgrind-light
+, withValgrind ? !stdenv.isDarwin && (stdenv.buildPlatform == stdenv.hostPlatform) && lib.meta.availableOn stdenv.hostPlatform valgrind-light, valgrind-light
```

Comment:
Looking at the code from `valgrind`, I think the issue is the static compilation actually.

Reply:
So I updated the code to check for `valgrind-light.meta.broken`. This should cover both cases: https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/tools/analysis/valgrind/default.nix#L102